### PR TITLE
Add example SAMDR i2c implementation

### PR DESCRIPTION
**Pull request by kelroy1990**
Implementation of the I2C on the SAMDR architecture with the ASF framework from ATMEL.

Regards, Carlos.

### CODE REVIEW

File: i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2018, Sensirion AG
```
Comment (rnestler) [inline i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c:2]:
```suggestion
 * Copyright (c) 2021, Sensirion AG
```

File: i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+			.hs_master_code  = 0x0,
+	    };
+		
+	    
+		    while (i2c_master_read_packet_wait(&i2c_master_instance, &packet) !=
+		    STATUS_OK) {
+			    /* Increment timeout counter and check if timed out if needed. */
+			    if (timeout++ == TIMEOUT) {
+					return -1;
+				    break;
```
Comment (rnestler) [inline i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c:127]:
```suggestion
```

A `break` after a return doesn't make sense.

**Review by rnestler — CHANGES_REQUESTED** [review]
Hi @kelroy1990

Thanks for the contribution! A few notes:
  * CI didn't seem to trigger since CircleCI seemed to be configured incorrectly. I changed it, so hopefully it will trigger next time you push
  * The code isn't formatted according to our style. Please use clang-format to fix it. (You may also use `make style-check`, which calls clang-format)
 * Please name the file sensirion_i2c_hal.c for consistency
  * Should this replace the `Atmel_SAMD2_series`? or is it really an additional implementation?

File: i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+/*
+*   Fixed I2C on SAMR architectures, this code its done to work with ASF from Microchip.
+*   15/11/2021
+*   On the i2c init, you should provide the CONF_MASTER_SDA_PINMUX and CONF_MASTER_SCK_PINMUX pin configuration.
+*   Then, we should configure the SERCOM module to use, with the CONF_I2C_MASTER_MODULE parameter, an example configuration for the ATSAMDR34J18 could be like:
+*   
+*   #define CONF_I2C_MASTER_MODULE    SERCOM3
+*   #define CONF_MASTER_SDA_PINMUX    PINMUX_PA16D_SERCOM3_PAD0
+*   #define CONF_MASTER_SCK_PINMUX    PINMUX_PA17D_SERCOM3_PAD1
+*   #define TIMEOUT 1000
```
Comment (rnestler) [inline i2c/sample-implementations/Atmel_SAMDR_series/sensirion_i2c_ASF.c:46]: Can you add
```
#ifndef CONF_I2C_MASTER_MODULE
#define CONF_I2C_MASTER_MODULE SERCOM3
#endif
```
for all the defines? That way it compiles by default, but a user can still override the defines.

**Comment by kelroy1990** [comment]
@rnestler all changes done

Regards, Carlos.

**Review by psachs — APPROVED** [review]

**Review by rnestler — APPROVED** [review]
I approve as well :slightly_smiling_face:

**Pull request merged**